### PR TITLE
Home13

### PR DESCRIPTION
**Pull request by IgorPrihodko**

### CODE REVIEW

File: src/main/java/homework13/task1/MinMax.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import java.util.Comparator;
+import java.util.function.BiConsumer;
+import java.util.stream.Stream;
+
+public class MinMax {
+    public static <T> void findMinMax(Stream<? extends T> stream, Comparator<? super T> order,
+                                      BiConsumer<? super T, ? super T> minMaxConsumer) {
+
+        T[] arr = (T[]) stream.sorted(order).toArray();
```
Comment (SevostyanovGerman) [inline src/main/java/homework13/task1/MinMax.java]: use min max

File: src/main/java/homework13/task2/Deserializable.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package homework13.task2;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+
+public class Deserializable {
+    public static Animal[] deserializeAnimalArray(byte[] data) {
+
+        Animal[] animals = null;
```
Comment (SevostyanovGerman) [inline src/main/java/homework13/task2/Deserializable.java]: why here?

**Review by SevostyanovGerman — CHANGES_REQUESTED** [review]

File: src/main/java/homework13/task1/MinMax.java (hunk 1 above)
Comment (IgorPrihodko) [reply src/main/java/homework13/task1/MinMax.java]: Done

File: src/main/java/homework13/task2/Deserializable.java (hunk 2 above)
Comment (IgorPrihodko) [reply src/main/java/homework13/task2/Deserializable.java]: fixed

File: src/main/java/homework13/task1/MinMax.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+public class MinMax {
+    public static <T> void findMinMax(Stream<? extends T> stream, Comparator<? super T> order,
+                                      BiConsumer<? super T, ? super T> minMaxConsumer) {
+
+        List<T> list = stream.sorted(order).collect(Collectors.toList());
+        if (list.size() == 0) {
+            minMaxConsumer.accept(null, null);
+        } else {
+            T max = list.stream().max(order).get();
```
Comment (SevostyanovGerman) [inline src/main/java/homework13/task1/MinMax.java]: why u use 2 streams? just use 1 stream object

**Review by SevostyanovGerman — CHANGES_REQUESTED** [review]

File: src/main/java/homework13/task1/MinMax.java (hunk 3 above)
Comment (IgorPrihodko) [reply src/main/java/homework13/task1/MinMax.java]: fixed

**Pull request merged**